src/common/provider.ts

```diff
@@ -1,10 +1,39 @@
-import { JsonRpcProvider } from '@ethersproject/providers';
+import { providers, utils } from 'ethers';
 import { ConfigService } from '@nestjs/config';
 import { Injectable } from '@nestjs/common';
 
+const { poll } = utils;
+
 @Injectable()
-export class Provider extends JsonRpcProvider {
+export class Provider extends providers.JsonRpcProvider {
+  private attempts: number;
+
   constructor(configService: ConfigService) {
     super(configService.get('ENS_URL'));
+    this.attempts = configService.get('CHAIN_REQUEST_ATTEMPTS');
+  }
+
+  /**
+   * Retries failed chain requests
+   * @override
+   */
+  public perform(method: string, params: any) {
```

Comment:
Is this mostly copied from the GitHub issue that you linked? If so, I think it is best to link to that issue here